pages/products/new.vue

```diff
@@ -0,0 +1,83 @@
+<script setup lang="ts">
+  import {validate} from "~/lib/validation";
+
+  const router = useRouter()
+  const product = reactive<ProductDTO>({
+    title: '',
+    description: '',
+    image: '',
+    price: '',
+  });
+
+  const image = ref('')
+
+  const errors = ref([])
+
+  const clearErrors = (path: string) => {
+    errors.value = errors.value.filter(e => e.path !== path && e.path !== 'server')
+  }
+
+  async function onSubmit() {
+    const validationErrors = validate(product)
+    if (validationErrors.length) {
+      errors.value = validationErrors
+      return
+    }
+
+    const {error} = await useFetch('/api/products', {
+      method: 'POST',
+      body: product
+    })
+
+    if (error.value) {
+      errors.value.push({path: 'server', message: error.value.data.statusMessage})
+      return
+    }
+
+    router.push({ path: "/products" })
+  }
+
+  const uploadImage = (e) => {
+
+    const image = e.target.files[0];
+    if (image.size / 1024 / 1024 > 1) {
+      alert('Image size should be less than 1MB')
+      return
+    }
+    const reader = new FileReader();
+    reader.readAsDataURL(image);
+    reader.onload = e => {
+      image.value = e.target.result;
+      product.image = image.value;
+    };
+  }
+</script>
+
+<template>
+
+  <div class="flex items-center justify-center min-w-80">
+    <div>
+      <NuxtLink to="/products" class="inline-block px-4 py-2 text-white">&#8592; Back</NuxtLink>
+      <p v-if="errors.find(e => e.path === 'server')" class="text-red-400">{{errors.find(e => e.path === 'server').message}}</p>
```

Comment:
This is definitely not the way to do this properly - adding `.find` into template. Why why shouldn't do this?

Reply:
In general, the error structure could have been designed better for these cases so you don't have to iterate over them.